src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -368,7 +368,13 @@ var DOMRenderer = ReactFiberReconciler({
       const hostContextDev = ((hostContext: any): HostContextDev);
       validateDOMNesting(null, text, null, hostContextDev.ancestorInfo);
     }
-    var textNode: TextInstance = document.createTextNode(text);
+
+    var doc = rootContainerInstance;
+    if (rootContainerInstance.nodeType !== DOCUMENT_NODE && doc.ownerDocument) {
```

Comment:
Also, is it really guaranteed that `rootContainerInstance` belongs to the same document? I would think that rendering an iframe and into it would break this assumption... or has this changed?

Reply:
From the way it WAS written before, Fiber would have broken the iframe case. If you looked at the ReactDOMFiberComponent's `createElement()`, you'd see that the semantics there for retrieving "ownerDocument" is the same as what I have written here:

```js
 var ownerDocument: Document = rootContainerElement.nodeType ===
      DOCUMENT_NODE
      ? (rootContainerElement: any)
      : rootContainerElement.ownerDocument;
```
I'm going to reuse this bit of code for clarity inside fiberentry